 Fix for wakelock being held during bootup and idle scenario.

On bootup of device, if device is kept idle without unlocking the
screen the TCXO does not happen due to "ActivityManager-Sleep"
partial wake_lock being held.

Fix releases the wakelock irrespective of the result of code flow
in checkReadyForSleepLocked() function.

diff --git a/services/core/java/com/android/server/am/ActivityStackSupervisor.java b/services/core/java/com/android/server/am/ActivityStackSupervisor.java
index ea565da..0f886a7 100644
--- a/services/core/java/com/android/server/am/ActivityStackSupervisor.java
+++ b/services/core/java/com/android/server/am/ActivityStackSupervisor.java
@@ -2773,6 +2773,9 @@ void goingToSleepLocked() {
             }
         }
         checkReadyForSleepLocked();
+        if (mGoingToSleep.isHeld()) {
+            mGoingToSleep.release();
+        }
     }
 
     boolean shutdownLocked(int timeout) {
@@ -2882,9 +2885,6 @@ void checkReadyForSleepLocked() {
 
         removeSleepTimeouts();
 
-        if (mGoingToSleep.isHeld()) {
-            mGoingToSleep.release();
-        }
         if (mService.mShuttingDown) {
             mService.notifyAll();
         }
